go/plugins/vertexai/modelgarden/anthropic/anthropic.go

```diff
@@ -0,0 +1,398 @@
+	req.MaxTokens = int64(MaxNumberOfTokens)
+
+	if c, ok := i.Config.(*ai.GenerationCommonConfig); ok && c != nil {
+		if c.MaxOutputTokens != 0 {
+			req.MaxTokens = int64(c.MaxOutputTokens)
+		}
+		if c.Version != "" {
+			req.Model = anthropic.Model(c.Version)
+		}
+		if c.Temperature != 0 {
+			req.Temperature = anthropic.Float(c.Temperature)
+		}
+		if c.TopK != 0 {
+			req.TopK = anthropic.Int(int64(c.TopK))
+		}
+		if c.TopP != 0 {
+			req.TopP = anthropic.Float(float64(c.TopP))
+		}
+		if len(c.StopSequences) > 0 {
+			req.StopSequences = c.StopSequences
+		}
+	}
+
+	// configure system prompt (if given)
+	sysBlocks := []anthropic.TextBlockParam{}
+	for _, message := range i.Messages {
+		if message.Role == ai.RoleSystem {
+			// only text is supported for system messages
+			sysBlocks = append(sysBlocks, anthropic.TextBlockParam{Text: message.Text()})
+		} else if message.Content[len(message.Content)-1].IsToolResponse() {
+			// if the last message is a ToolResponse, the conversation must continue
+			// and the ToolResponse message must be sent as a user
+			// see: https://docs.anthropic.com/en/docs/build-with-claude/tool-use#handling-tool-use-and-tool-result-content-blocks
+			parts, err := convertParts(message.Content)
+			if err != nil {
+				return req, err
+			}
+			messages = append(messages, anthropic.NewUserMessage(parts...))
+		} else {
+			parts, err := convertParts(message.Content)
+			if err != nil {
+				return req, err
+			}
+			messages = append(messages, anthropic.MessageParam{
+				Role:    toAnthropicRole(message.Role),
+				Content: parts,
+			})
+		}
+	}
+
+	req.System = sysBlocks
+	req.Messages = messages
+
+	tools, err := convertTools(i.Tools)
+	if err != nil {
+		return req, err
+	}
+	req.Tools = tools
+
+	return req, nil
+}
+
+// convertTools translates [ai.ToolDefinition] to an anthropic.ToolParam type
+func convertTools(tools []*ai.ToolDefinition) ([]anthropic.ToolUnionParam, error) {
+	resp := make([]anthropic.ToolUnionParam, 0)
+	regex := regexp.MustCompile(ToolNameRegex)
+
+	for _, t := range tools {
+		if t.Name == "" {
+			return nil, fmt.Errorf("tool name is required")
+		}
+		if !regex.MatchString(t.Name) {
+			return nil, fmt.Errorf("tool name must match regex: %s", ToolNameRegex)
+		}
+
+		resp = append(resp, anthropic.ToolUnionParam{
+			OfTool: &anthropic.ToolParam{
+				Name:        t.Name,
+				Description: anthropic.String(t.Description),
+				InputSchema: generateSchema[map[string]any](),
+			},
+		})
+	}
+
+	return resp, nil
+}
+
+func generateSchema[T any]() anthropic.ToolInputSchemaParam {
+	reflector := jsonschema.Reflector{
+		AllowAdditionalProperties: true,
+		DoNotReference:            true,
+	}
+	var v T
+	schema := reflector.Reflect(v)
+	return anthropic.ToolInputSchemaParam{
+		Properties: schema.Properties,
+	}
+}
+
+// convertParts translates [ai.Part] to an anthropic.ContentBlockParamUnion type
+func convertParts(parts []*ai.Part) ([]anthropic.ContentBlockParamUnion, error) {
```

Comment:
```suggestion
func toAnthropicParts(parts []*ai.Part) ([]anthropic.ContentBlockParamUnion, error) {
```

Follow this pattern elsewhere, like in `googlegenai`.